### PR TITLE
Add docs for cylc clean

### LINKED ISSUE #256: Document cylc clean
https://github.com/cylc/cylc-flow/issues/3887

### PR DESCRIPTION
**Pull request by MetRonnie**
Close #256

### CODE REVIEW

**Review by oliver-sanders — APPROVED** [review]

File: src/user-guide/removing-workflows.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+==================
+
+If you want to delete an installed workflow from the :term:`cylc-run directory`
+(or just delete certain subdirectories), the recommended way is using
+the ``cylc clean`` command. ``cylc clean`` takes care of deleting workflows
+on the local filesystem was well as any remote install targets. It also
+follows any symlink directories specified in
+:cylc:conf:`global.cylc[install][symlink dirs]`
+(but will never follow other symlink directories). It will not remove the
+workflow :term:`source directory` either.
```
Comment (wxtim) [inline src/user-guide/removing-workflows.rst:13]: A couple of points - 
- Hemmingway Editor gave this a US reading grade of 16, I managed to get it down to 11 with

> To delete an installed workflow (or run) from the cylc run directory use cylc clean. Cylc clean removes files from remote install targets as well as local file systems. It also follows any symlink directories specified in global.cylc[install][symlink dirs] (but will never follow other symlink directories). It does not remove the workflow source directory either.

- I couldn't streamline the sentence sentence starting "It also follows any..."
- At the risk of pointing my own flaws, I think all sentences here should be in the same tense. I'd suggest the "will" in the last sentence be "does"

File: src/user-guide/removing-workflows.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+If you've used ``rose suite-clean`` before, the functionality is similar, but
```
Comment (wxtim) [inline src/user-guide/removing-workflows.rst:15]: could be in a ..note?

File: src/user-guide/removing-workflows.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+not identical.
+
+``cylc clean`` can be called like so:
```
Comment (wxtim) [inline src/user-guide/removing-workflows.rst:18]:
```suggestion
Simple ``cylc clean`` example
-----------------------------

```

File: src/user-guide/removing-workflows.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+----------------------------------
+
+If any jobs in your workflow ran on a remote :term:`platform`, Cylc will
+automatically remove the workflow files on there in addition to the local
+filesystem.
+
+.. code-block:: console
+
+   $ cylc clean remote-example
+   INFO - Cleaning on install target: enterprise1701
```
Comment (wxtim) [inline src/user-guide/removing-workflows.rst:43]: LOL

**Review by wxtim — APPROVED** [review]
Nothing fatal. Some things to consider.

File: src/user-guide/removing-workflows.rst (hunk 2 above)
Comment (MetRonnie) [reply src/user-guide/removing-workflows.rst:15]: I feel like we use `.. note`s a lot!

**Review by wxtim — APPROVED** [review]

**Comment by wxtim** [comment]
1 Reviewer becuase:
- Docs
- Oliver approved before I asked for changes

**Pull request merged**